packages/engine-server/src/util/inMemoryNoteCache.ts

```diff
@@ -0,0 +1,19 @@
+import { NoteProps } from "@dendronhq/common-all";
+import { InMemoryNoteCacheImpl } from "./impl/inMemoryNoteCacheImpl";
+
+/**
+ * In memory cache for {@link NoteProps}.
+ *
+ * Use {@link InMemoryNoteCacheFactory} to get an instance of this interface.*/
+export interface InMemoryNoteCache {
```

Comment:
I would actually call this something more generic - nothing about the interface methods require it to be either In-memory or a cache. Then, we could rename `InMemoryNoteCacheImpl` -> `InMemoryNoteCache`, which is the in-memory cache implementation of the `NoteRetriever` or whatever name we settle on.